tests/core/test_axes.py

```diff
@@ -1,182 +1,149 @@
+        })
+    simulation_builder.register_variable('rent', group_entity)
+    simulation_builder.add_parallel_axis({'count': 2, 'name': 'rent', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert [role.key for role in simulation_builder.get_roles('households')] == ['parent', 'child', 'parent', 'parent', 'child', 'parent']
+
+
+def test_add_axis_on_persons_distributes_roles(simulation_builder, persons, group_entity):
+    simulation_builder.add_person_entity(persons, {'Alicia': {}, 'Javier': {}, 'Tom': {}})
+    simulation_builder.add_group_entity('persons', ['Alicia', 'Javier', 'Tom'], group_entity, {
+        'housea': {'parents': ['Alicia']},
+        'houseb': {'parents': ['Tom'], 'children': ['Javier']},
+        })
+    simulation_builder.register_variable('salary', persons)
+    simulation_builder.add_parallel_axis({'count': 2, 'name': 'salary', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert [role.key for role in simulation_builder.get_roles('households')] == ['parent', 'child', 'parent', 'parent', 'child', 'parent']
+
+
+def test_add_axis_distributes_memberships(simulation_builder, persons, group_entity):
+    simulation_builder.add_person_entity(persons, {'Alicia': {}, 'Javier': {}, 'Tom': {}})
+    simulation_builder.add_group_entity('persons', ['Alicia', 'Javier', 'Tom'], group_entity, {
+        'housea': {'parents': ['Alicia']},
+        'houseb': {'parents': ['Tom'], 'children': ['Javier']},
+        })
+    simulation_builder.register_variable('rent', group_entity)
+    simulation_builder.add_parallel_axis({'count': 2, 'name': 'rent', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert simulation_builder.get_memberships('households') == [0, 1, 1, 2, 3, 3]
+
+
+def test_add_perpendicular_axes(simulation_builder, persons):
+    simulation_builder.add_person_entity(persons, {'Alicia': {}})
+    simulation_builder.register_variable('salary', persons)
+    simulation_builder.register_variable('pension', persons)
+    simulation_builder.add_parallel_axis({'count': 3, 'name': 'salary', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.add_perpendicular_axis({'count': 2, 'name': 'pension', 'min': 0, 'max': 2000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert simulation_builder.get_input('salary', '2018-11') == approx([0, 1500, 3000, 0, 1500, 3000])
+    assert simulation_builder.get_input('pension', '2018-11') == approx([0, 0, 0, 2000, 2000, 2000])
+
+# Integration test
+
+
+def test_simulation_with_axes(simulation_builder):
+    from .test_countries import tax_benefit_system
+    input_yaml = """
+        persons:
+          Alicia: {salary: {2018-11: 0}}
+          Javier: {}
+          Tom: {}
+        households:
+          housea:
+            parents: [Alicia, Javier]
+          houseb:
+            parents: [Tom]
+        axes:
```

Comment:
I'm curious what brought us to mix axes and the `build_from_dict` method. I was more imagining that we would make a separate method, such as `build_from_axes`, as I think instantiating a simulation from axes and building one from JSON/YAML are pretty distinct use cases. Is it because there is a lot of common logics between both cases?

Reply:
You can't really build a simulation only from axes, rather axes are a way to replicate an already constructed situation (which is convenient to do using the dictionary format) a certain number of times, varying one or more input variables along the way. You have to have a situation in place already.

Reply:
Ok, thanks for the explanation!